### ISSUE TITLE
ICS ?: IBC Encoding Scheme

### ISSUE THREAD

**Issue by mossid** [cosmos/ibc#28]
Will Cover:
* Requirements for the encoding schemes to encode packets, metadata, Merkle proofs, and any other IBC related data that has to be transferred between chains
* Initially supported encoding schemes, candidates: amino-binary, proto3, amino-json, abi
* Common wire format language to describe data types(optional)
* Interpreter between common wire format and concrete encoding schemes(optional)

**Comment by cwgoes** [comment]
Should this be its own ICS or should encoding schemes for relevant data structures be included in their respective ICSs (e.g. packet, connection, channel)?

**Comment by mossid** [comment]
I think both, the concrete definition for each data structure should be included in their respective ics, but the encoding schemes themselves those IBC will support(and requirements for them) should be written in a separated ics. For example, in this way, we can add a new encoding scheme to IBC without modifying each ICSs.

Maybe I'm using the wrong term, is "wire protocol" more appropriate?

**Comment by jackzampolin** [comment]
Protobuf please! 🙏

**Comment by cwgoes** [comment]
I agree that we should separate out standardized wire protocol(s) into individual ICSs.

I think we should avoid constructing proprietary wire protocols unless we have an extremely strong reason to do so.

Wire protocols we might want to support in IBC:
- Proto3 (recommended)
- RLP

**Comment by cwgoes** [comment]
This no longer needs to be a separate spec, concluded as above.

IBC 1.0.0 will include canonical proto3 schema files for appropriate data structures.

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
